lib/stm32/common/usart_common_all.c

```diff
@@ -45,11 +45,15 @@ rcc_clock_setup-* functions in RCC).
 Note: For LPUART, baudrates over 2**24 (~16.7 Mbaud) may overflow
 the calculation and are therefore not supported by this function.
 
+Note: This is not compatible with the complex clock tree implemented by the H7
+series of parts, and is defined separately for those modules.
+
 @param[in] usart unsigned 32 bit. USART block register address base @ref
 usart_reg_base
 @param[in] baud unsigned 32 bit. Baud rate specified in Hz.
 */
 
+#ifndef STM32H7
```

Comment:
yeah, we're not goign to do this.  If there's a lot of other common code we do want, fine, then this goes somewhere else, but we're not goign to start putting ifndefs into this file.